nexus/src/db/datastore/project.rs

```diff
@@ -65,33 +66,241 @@ impl DataStore {
         })
     }
 
+    async fn ensure_no_instances_in_project(
```

Comment:
Feels like a lot of boilerplate around these functions but I don't have any particular suggestions to improve it.

Reply:
Done in 0249d63ed7b7317167ab8bfb961bfd87ca603ae8 via a macro; I'm not sure if this is better or worse